src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -605,6 +594,9 @@ public override TResult[] ToArray()
         public static IEnumerable<TSource> Take<TSource>(this IEnumerable<TSource> source, int count)
         {
             if (source == null) throw Error.ArgumentNull("source");
+            if (count <= 0) return EmptyPartition<TSource>.Instance;
```

Comment:
Or perhaps Empty<TSource>() would be better. Since `GetEnumerator()` isn't hit in the original form, there's no meaningful observable difference this way, but there could be with Empty<TSource>().